pkg/client/events/options.go

```diff
@@ -9,3 +9,18 @@ func WithDialer(dialer client.Dialer) client.EventsQueryClientOption {
 		evtClient.(*eventsQueryClient).dialer = dialer
 	}
 }
+
+// WithConnRetryLimit returns an option function which sets the number
+// of times the replay client should retry in the event that it encounters
+// an error or its connection is interrupted.
+// If connRetryLimit is < 0, it will retry indefinitely.
+func WithConnRetryLimit[T any](limit int) client.EventsReplayClientOption[T] {
+	return func(client client.EventsReplayClient[T]) {
+		// Ignore the zero value because limit may be provided via a partially
```

Comment:
@bryanchriswhite When you work on updating one governance parameter at a time, taking this pattern into consideration will be critical!